src/NuGet.Clients/NuGet.VisualStudio.Common/SolutionRestore/SolutionRestoreRequest.cs

```diff
@@ -32,6 +36,16 @@ public sealed class SolutionRestoreRequest
             RestoreSource = restoreSource;
         }
 
+        public SolutionRestoreRequest(
```

Comment:
Everything should consolidate to the same constructor. Don't set the values in multiple constructors as that can be error prone. The constructor with 2 params should call the constructor with 2 params.